### PR TITLE
U4X-605: Integration of Facility Registry with UgandaEMR Plus

### PR DESCRIPTION
**Pull request by Daphne210**
https://metsprogramme.atlassian.net/browse/U4X-605

https://github.com/METS-Programme/esm-ugandaemr-core/assets/89844768/b6673651-97e3-4242-91b1-12879f7630ab

### CODE REVIEW

File: packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx [hunk 1]
```diff
@@ -13,23 +13,21 @@ import {
...
 interface FacilityCodeDetails {
   value?: string;
-  uuid?: string;
 }
 
 const OverallSystemInfo = ({ buildInfo, emrVersion, facilityCodeDetails, setFacilityCodeDetails }) => {
   const { t } = useTranslation();
   const buildDateTime =
-    Object.keys(buildInfo).length > 0
+    buildInfo && buildInfo['SystemInfo.OpenMRSInstallation.systemDate']
```
Comment (jabahum) [inline packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx:30]: move this to config or constants

File: packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx [hunk 2]
```diff
@@ -142,47 +135,52 @@ const SystemInfoPage = () => {
...
     }
-  }, [facilityCodeDetails, facilityIds]);
+  }, [facilityIds]);
 
-  useEffect(() => {
-    if (facilityCodeDetails.uuid !== null && facilityCodeDetails.value !== null) {
-      updatePropertyValue(facilityCodeDetails.uuid, facilityCodeDetails.value).then(
+  const updateFacilityCode = useCallback(() => {
+    if (facilityCodeDetails.value) {
+      updatePropertyValue('ugandaemrsync.national.health.facility.registry.identifier', facilityCodeDetails.value).then(
```
Comment (jabahum) [inline packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx:152]: this too

File: packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx (hunk 1 above)
Comment (Daphne210) [reply packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx:30]: > move this to config or constants

This has been addressed in the subsequent commit

File: packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx (hunk 2 above)
Comment (Daphne210) [reply packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx:152]: > this too

This has also been addressed in the subsequent commit

**Review by jabahum — APPROVED** [review]
Thanks @Daphne210

**Pull request merged**